Reason: Slower platforms (everything except x86_64 and aarch64) run into a timeout in the CI
Upstream: No

diff --git a/internal/warpc/warpc.go b/internal/warpc/warpc.go
index be8e48bb5..47ac2e472 100644
--- a/internal/warpc/warpc.go
+++ b/internal/warpc/warpc.go
@@ -234,7 +234,13 @@ func (p *dispatcherPool[Q, R]) Execute(ctx context.Context, q Message[Q]) (Messa
 		return d.zeroR, err
 	}
 
-	timer := getTimer(30 * time.Second)
+	timeout := 30 * time.Second
+	if runtime.GOARCH == "riscv64" {
+		timeout *= 20
+	} else if runtime.GOARCH != "amd64" && runtime.GOARCH != "arm64" {
+		timeout *= 10
+	}
+	timer := getTimer(timeout)
 	defer putTimer(timer)
 
 	select {
